plan/refiner.go

```diff
@@ -44,10 +45,9 @@ func refine(in Plan) error {
 	case *TableScan:
 		err = buildTableRange(x)
 	case *NewTableScan:
-		x.Ranges = []TableRange{{math.MinInt64, math.MaxInt64}}
-		return nil
+		err = buildNewTableRange(x)
```

Comment:
Get conditions from the parent `*Selection` of `*NewTableScan`, then build new table range.